src/app/modules/inventory/components/sections/create-section/create-section.component.ts

```diff
@@ -0,0 +1,208 @@
+        {
+          next: (response: Section[]) =>{
+            this.sections = response;
+          },
+          error: (err) => {
+            console.log(err);
+          }
+        }
+      )
+    );
+  }
+
+  saveSection(form: NgForm){
+    if(this.editing) this.editSection(form)
+    else this.createSection(form);
+  }
+
+  createSection(form: NgForm){
+    if(form.invalid){
+      alert('Formulario invalido');
+      return;
+    }
+
+    if(this.sections.some(section => section.name == this.newSection.name && section.id_zona ==
+      this.newSection.id_zona)){
+      alert("Ya existe una seccion con ese nombre y zona");
+      return;
+    }
+
+    this.sectionService.create(this.newSection).subscribe(
+      {
+        next: (response: Section) =>{
+          alert("Sección creada!");
+          this.sections.push(response);
+          this.newSection.name = "";
+          this.newSection.id_zona = 0;
+          this.filtrar();
+        },
+        error: (err) => {
+          alert("Hubo un error al intentar esta operación, no se pudo crear la sección")
+        }
+      }
+    );
+  }
+
+  editSection(form: NgForm){
+    if(form.invalid){
+      alert('Formulario invalido');
+      return;
+    }
+
+    if(this.sections.some(section => section.name == this.newSection.name && section.id_zona ==
+      this.newSection.id_zona)){
+      alert("Ya existe una seccion con ese nombre y zona");
+      return;
+    }
+
+    this.sectionService.edit(this.newSection).subscribe(
+      {
+        next: (response: Section) =>{
+          alert("Sección editada!");
+          this.sections = this.sections.filter(section => { section.id != this.newSection.id })
+          this.newSection.name = "";
+          this.newSection.id_zona = 0;
+          this.filtrar();
+          this.editing = false;
+        },
+        error: (err) => {
+          alert("Hubo un error al intentar esta operación, no se pudo editar la sección")
+        }
+      }
+    );
+  }
+
+  ngOnDestroy(): void {
+    this.subscriptions.unsubscribe();
+  }
+
+  filtrar(){
+    this.filtrando = true;
+    this.subscriptions.add(
+      this.sectionService.getByZone(this.sectionParam.id_zona).subscribe(
+        {
+          next: (response: Section[]) =>{
+            this.sectionsFiltered = response;
+          },
+          error: (err) => {
+            console.log(err);
+          }
+        }
+      )
+    )
+    this.storageZones.forEach(zone => {
+      if(zone.id == this.sectionParam.id_zona){
+        this.zoneFiltered.name = zone.name;
+      }
+    });
+  }
+
+  eliminar(id: number){
+    let borrar = confirm("seguro que desea eliminar la sección?");
+    if(!borrar) return;
+
+    let flag:Boolean = false;
+    try{
+      this.batchs.forEach(lote => {
+        if(lote.id_section == id){
+          flag = true;
+          alert("No se puede eliminar esta sección porque hay un lote que depende de ella")
+          throw "Existe un lote que depende de esta sección";
```

Comment:
Para cortar la ejecución del código se podría utilizar un return en vez de un throw exception

Reply:
El return no me estaría deteniendo el forEach()

Reply:
puede ser que sea debido a que es una arrow function. Proba con break y si no te funciona podes probar cambiado a un for tradicional en vez de ese operador de arreglos.